fleetspeak/src/e2etesting/lib/setup_components.go

```diff
@@ -0,0 +1,296 @@
+package main
+
+import (
+	"context"
+	"flag"
+	"fmt"
+	"github.com/golang/protobuf/proto"
+	ptypes "github.com/golang/protobuf/ptypes"
+	duration "github.com/golang/protobuf/ptypes/duration"
+	daemonservice_pb "github.com/google/fleetspeak/fleetspeak/src/client/daemonservice/proto/fleetspeak_daemonservice"
+	client_config_pb "github.com/google/fleetspeak/fleetspeak/src/client/generic/proto/fleetspeak_client_generic"
+	"github.com/google/fleetspeak/fleetspeak/src/common"
+	spb "github.com/google/fleetspeak/fleetspeak/src/common/proto/fleetspeak"
+	cpb "github.com/google/fleetspeak/fleetspeak/src/config/proto/fleetspeak_config"
+	fcpb "github.com/google/fleetspeak/fleetspeak/src/server/components/proto/fleetspeak_components"
+	grpcservice_pb "github.com/google/fleetspeak/fleetspeak/src/server/grpcservice/proto/fleetspeak_grpcservice"
+	services_pb "github.com/google/fleetspeak/fleetspeak/src/server/proto/fleetspeak_server"
+	"google.golang.org/grpc"
+	"io"
+	"io/ioutil"
+	"os"
+	"os/exec"
+	"path"
+	"sync"
+	"syscall"
+	"time"
+)
+
+var (
+	mysql_database = flag.String("mysql_database", "", "MySQL database name to use")
+	mysql_username = flag.String("mysql_username", "", "MySQL username to use")
+	mysql_password = flag.String("mysql_password", "", "MySQL password to use")
+)
+
+type startedProcessesPids struct {
+	server_pid  chan int
+	client_pid  chan int
+	service_pid chan int
+}
+
+func (sp *startedProcessesPids) init() {
+	sp.server_pid = make(chan int, 1)
+	sp.client_pid = make(chan int, 1)
+	sp.service_pid = make(chan int, 1)
+}
+
+func killProcess(pid int) {
+	syscall.Kill(pid, syscall.SIGTERM)
+	for {
+		process, err := os.FindProcess(pid)
+		if err != nil {
+			continue
+		}
+		err = process.Signal(syscall.Signal(0))
+	}
+}
+
+func (sp *startedProcessesPids) killAll() {
+	pids := make([]int, 0)
+	close(sp.server_pid)
+	close(sp.client_pid)
+	close(sp.service_pid)
+	for pid := range sp.server_pid {
+		pids = append(pids, pid)
+	}
+	for pid := range sp.client_pid {
+		pids = append(pids, pid)
+	}
+	for pid := range sp.service_pid {
+		pids = append(pids, pid)
+	}
+
+	for _, pid := range pids {
+		syscall.Kill(pid, syscall.SIGTERM)
+	}
+	for {
+		finished := true
+		for _, pid := range pids {
+			process, err := os.FindProcess(pid)
+			if err != nil {
+				continue
+			}
+			err = process.Signal(syscall.Signal(0))
+			if err == nil {
+				finished = false
+				break
+			} else {
+				fmt.Println(pid, err)
```

Comment:
This has to o to stderr. The error message should be more descriptive ("Failed to kill process <pid>: <error>")